pkg/kube/inject/webhook.go

```diff
@@ -575,6 +574,25 @@ func reapplyOverwrittenContainers(finalPod *corev1.Pod, originalPod *corev1.Pod,
 	return finalPod, nil
 }
 
+func resetFieldsInAutoImageContainer(original *corev1.Container, template *corev1.Container) {
+	if original.Image != AutoImage {
+		return
+	}
+
+	original.Image = ""
+
+	// If the original pod comes with SecurityContext.RunAsUser and the template defines a value different than the default (1337),
+	// then ignore the original value and stick with the final (merged one)
+	// This is likely a scenario in OpenShift when the istio-proxy container with image: auto is parsed, if SecurityContext.RunAsUser
+	// does not exist, OpenShift automatically assigns a value which is based on an annotation in the namespace. Regardless if the user
+	// provided that value or if it was assigned by OpenShift, the correct value is the one in the template, as set by the `.ProxyUID` field.
+	if original.SecurityContext != nil && template.SecurityContext != nil && template.SecurityContext.RunAsUser != nil &&
+		*template.SecurityContext.RunAsUser != constants.DefaultProxyUIDInt {
```

Comment:
@jwendell Does not it update other platform?

Reply:
Is this from the injector template? If so this has no influence

Reply:
If the value coming from the template is different than 1337, it means it is OpenShift, thus it must override what the user defines. If the value is the default 1337 (meaning regular k8s), this is a no-op (as the change in the test file proves).

Reply:
make sense to me